src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -136,3 +151,39 @@ def _inner_not_found(ex):
 
     with self.command_group('cdn edge-node', cdn_edge_sdk) as g:
         g.command('list', 'list')
+
+    with self.command_group('cdn waf policy', cdn_waf_policy_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list')
+        g.custom_command('set', 'set_waf_policy', client_factory=cf_waf_policy)
+        g.command('delete', 'delete')
+
+    with self.command_group('cdn waf policy managed-rule-set', cdn_waf_policy_sdk, is_preview=True) as g:
+        g.custom_command('add', 'add_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('remove', 'remove_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('list', 'list_waf_policy_managed_rule_sets', client_factory=cf_waf_policy)
+        g.custom_show_command('show', 'show_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('list-available', 'list_waf_managed_rule_set', client_factory=cf_waf_rule_set)
+        g.custom_command('show-available', 'show_waf_managed_rule_set', client_factory=cf_waf_rule_set)
+
+    with self.command_group('cdn waf policy managed-rule-set rule-group-override',
+                            cdn_waf_policy_sdk,
+                            is_preview=True) as g:
+        g.custom_command('set', 'set_waf_managed_rule_group_override', client_factory=cf_waf_policy)
+        g.custom_command('delete', 'delete_waf_managed_rule_group_override', client_factory=cf_waf_policy)
+        g.custom_command('list', 'list_waf_policy_managed_rule_group_overrides', client_factory=cf_waf_policy)
+        g.custom_show_command('show', 'show_waf_managed_rule_group_override', client_factory=cf_waf_policy)
+        g.custom_command('list-available', 'list_waf_managed_rule_groups', client_factory=cf_waf_rule_set)
+        g.custom_command('show-available', 'show_waf_managed_rule_group', client_factory=cf_waf_rule_set)
```

Comment:
The same as above

Reply:
Same as above, please let me know if there is any issue here and I'll just remove it for now to unblock this PR.

Reply:
Is there any special design for rule-group-override? What is the difference between override and standard rule group?

Reply:
Managed rules (including managed rule sets and rule groups) are defined by our service, and cannot be modified. The default settings of a rule group can be overridden for a particular policy, but this does not change the same managed rules which may be applied to a different policy. For this reason, the service refers to these as Rule Group Overrides, and I kept this same convention for our CLI module.